operations/app/src/modules/sftp_container/main.tf

```diff
@@ -12,12 +12,26 @@ resource "azurerm_network_profile" "sftp_network_profile" {
   }
 }
 
+resource "azurerm_network_profile" "sftp_vnet_network_profile" {
+  name                = "sftp_vnet_network_profile"
+  location            = var.location
+  resource_group_name = var.resource_group
+
+  container_network_interface {
+    name = "sftp_container_vnet_network_interface"
+    ip_configuration {
+      name      = "sftp_container_vnet_ip_configuration"
+      subnet_id = data.azurerm_subnet.container_subnet.id
+    }
+  }
+}
+
```

Comment:
A separate network profile needed to be created as Terraform was confused by just updating the subnet_id.